src/domains/user/accounts.routes.integration.spec.js

```diff
@@ -18,10 +18,122 @@ describeDbTestSuite('accountsRouter', () => {
+      };
+
+      const response = await request(app)
+        .post('/signup')
+        .set('Content-type', 'application/json')
+        .send(createAccountBody);
+      const { message } = response.body;
+      expect(response.body).toEqual({
+        message,
+      });
+      expect(response.status).toEqual(400);
+    });
+
+    test('it should not create an account without password', async () => {
+      const createAccountBody = {
+        email: 'email@example.com',
+      };
+
+      const response = await request(app)
+        .post('/signup')
+        .set('Content-type', 'application/json')
+        .send(createAccountBody);
+      const { message } = response.body;
+      expect(response.body).toEqual({
+        message,
+      });
+      expect(response.status).toEqual(400);
+    });
+
+    test('it should not create an account with invalid email', async () => {
+      const createAccountBody = {
+        email: 'emailexample.com',
+        password: '0PasSwoRd12',
+      };
+
+      const response = await request(app)
+        .post('/signup')
+        .set('Content-type', 'application/json')
+        .send(createAccountBody);
+      const { message } = response.body;
+      expect(response.body).toEqual({
+        message,
+      });
+      expect(response.status).toEqual(400);
+    });
+
+    test('it should not create an account with short password', async () => {
+      const createAccountBody = {
+        email: 'email@example.com',
+        password: '0Pas',
+      };
+
+      const response = await request(app)
+        .post('/signup')
+        .set('Content-type', 'application/json')
+        .send(createAccountBody);
+      const { message } = response.body;
+      expect(response.body).toEqual({
+        message,
+      });
+      expect(response.status).toEqual(400);
+    });
+
+    test('It should return invalid when login with wrong credential', async () => {
+      const credentials = {
+        email: 'wrong@example.com',
+        password: 'wrongpass',
+      };
+
+      const response = await request(app)
+        .post('/login')
+        .set('Content-type', 'application/json')
+        .send(credentials);
+      const { message } = response.body;
+      expect(response.body).toEqual({
+        message,
+      });
+      expect(response.status).toEqual(400);
+    });
+
+    test('it should login user successfully', async () => {
+      const createAccountBody = {
+        email: 'email@example.com',
+        password: '0PasSwoRd12',
+      };
+      await request(app)
+        .post('/signup')
+        .set('Content-type', 'application/json')
+        .send(createAccountBody);
+
+      const credentials = {
+        email: 'email@example.com',
+        password: '0PasSwoRd12',
+      };
+
+      const response = await request(app)
+        .post('/login')
+        .set('Content-type', 'application/json')
+        .send(credentials);
+      const { message, token } = response.body;
```

Comment:
since `token` is indeterministic, you can assert that the coming response has a token.

`expect(response.body).toHaveProperty('token')`

since we can't assert the exact value of the token. However you should assert the exact value of the message as we know what message the app should return.

Reply:
Awesome